Kernel/Devices/PATAChannel.cpp

```diff
@@ -98,34 +100,31 @@ PATAChannel::PATAChannel(ChannelType type)
     m_dma_enabled.resource() = true;
     ProcFS::add_sys_bool("ide_dma", m_dma_enabled);
 
-    initialize();
+    initialize(force_pio);
     detect_disks();
 }
 
 PATAChannel::~PATAChannel()
 {
 }
 
-void PATAChannel::initialize()
+void PATAChannel::initialize(bool force_pio)
 {
-    static const PCI::ID piix3_ide_id = { 0x8086, 0x7010 };
-    static const PCI::ID piix4_ide_id = { 0x8086, 0x7111 };
     PCI::enumerate_all([this](const PCI::Address& address, PCI::ID id) {
-        if (id == piix3_ide_id || id == piix4_ide_id) {
```

Comment:
This is a very important change and it should be mentioned in the commit message.
After this patch, we're no longer limited to PIIX3 and PIIX4 chipsets for DMA.

Reply:
> This is a very important change and it should be mentioned in the commit message.
> After this patch, we're no longer limited to PIIX3 and PIIX4 chipsets for DMA.
I agree, and I changed the commit message :)